### PR TITLE
Add gz-sim-yarp-plugins package

### PR DESCRIPTION
**Pull request by xela-95**
This PR adds the https://github.com/robotology/gz-sim-yarp-plugins package.

Reference issue: https://github.com/robotology/gz-sim-yarp-plugins/issues/131

### CODE REVIEW

**Comment by xela-95** [comment]
@traversaro I was looking how to enable the build of `gz-sim-yarp-plugins` project to test everything works. This comes with the new option `ROBOTOLOGY_USES_GZ` that is disabled by default and mutually exclusive wrt `ROBOTOLOGY_USES_GAZEBO`.

Should I configure an additional step similar to
https://github.com/robotology/robotology-superbuild/blob/5737298429d6a781c49bdc5eba0dc151b0ec9f77/.github/workflows/ci.yml#L384-L391

with the `ROBOTOLOGY_USES_GZ` option set to ON?

**Comment by traversaro** [comment]
Ah, this thing of mutual exclusivity is complex and is going to be an headache. Until now, the logic in the CI was always: "just enable everything". Now I guess we need either to:
* Have different builds for `ROBOTOLOGY_USES_GZ` and `ROBOTOLOGY_USES_GAZEBO`, doubling the build matrix
* At the end of the build, add a step that enables  `ROBOTOLOGY_USES_GZ` and disables `ROBOTOLOGY_USES_GAZEBO` (or the contrary)

Feel free to follow the path that make more sense to you, probably the second one is easier.

**Comment by xela-95** [comment]
@traversaro in https://github.com/robotology/robotology-superbuild/blob/5737298429d6a781c49bdc5eba0dc151b0ec9f77/doc/conda-forge.md?plain=1#L181-L201 is not mentioned that when configuring the project the `CMAKE_INSTALL_PREFIX` variable should be set to `CONDA_PREFIX`. Is it on purpose?

**Comment by traversaro** [comment]
The superbuild is a bit special, as it does not have any install step. In its `build` (i.e. the invocation of `make` or `ninja`) step, it calls the install of all the subprojects. All the subprojects are installed in the `build/install` location, and this can't be easily changed. To consume projects compiled by the superbuild, one should source `setup.sh` script or similar, not install  in `$CONDA_PREFIX`.

**Comment by xela-95** [comment]
> The superbuild is a bit special, as it does not have any install step. In its `build` (i.e. the invocation of `make` or `ninja`) step, it calls the install of all the subprojects. All the subprojects are installed in the `build/install` location, and this can't be easily changed. To consume projects compiled by the superbuild, one should source `setup.sh` script or similar, not install in `$CONDA_PREFIX`.

Ok it's clear now thanks!

**Comment by xela-95** [comment]
I'm trying to perform the installation of the superbuild with [conda dependencies](https://github.com/robotology/robotology-superbuild/blob/master/doc/conda-forge.md#source-installation). When I build the project I get the following error:

```
[33/52] Performing configure step for 'gz-sim-yarp-plugins'
FAILED: src/gz-sim-yarp-plugins/CMakeFiles/YCMStamp/gz-sim-yarp-plugins-configure /home/acroci/repos/robotology-superbuild/build/src/gz-sim-yarp-plugins/CMakeFiles/YCMStamp/gz-sim-yarp-plugins-configure 
cd /home/acroci/repos/robotology-superbuild/build/src/gz-sim-yarp-plugins && /home/acroci/mambaforge/envs/robsub/bin/cmake --no-warn-unused-cli "-DCMAKE_PREFIX_PATH:PATH=/home/acroci/repos/robotology-superbuild/build/install;/home/acroci/mambaforge/envs/robsub;/home/acroci/mambaforge/envs/robsub/x86_64-conda-linux-gnu/sysroot/usr" -C/home/acroci/repos/robotology-superbuild/build/src/gz-sim-yarp-plugins/CMakeFiles/YCMTmp/gz-sim-yarp-plugins-cache-Release.cmake -GNinja /home/acroci/repos/robotology-superbuild/src/gz-sim-yarp-plugins && /home/acroci/mambaforge/envs/robsub/bin/cmake -E touch /home/acroci/repos/robotology-superbuild/build/src/gz-sim-yarp-plugins/CMakeFiles/YCMStamp/gz-sim-yarp-plugins-configure
Not searching for unused variables given on the command line.
loading initial cache file /home/acroci/repos/robotology-superbuild/build/src/gz-sim-yarp-plugins/CMakeFiles/YCMTmp/gz-sim-yarp-plugins-cache-Release.cmake
-- The C compiler identification is GNU 12.3.0
-- The CXX compiler identification is GNU 12.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/acroci/mambaforge/envs/robsub/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/acroci/mambaforge/envs/robsub/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:5 (find_package):
  By not providing "Findgz-cmake3.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "gz-cmake3", but CMake did not find one.

  Could not find a package configuration file provided by "gz-cmake3" with
  any of the following names:

    gz-cmake3Config.cmake
    gz-cmake3-config.cmake

  Add the installation prefix of "gz-cmake3" to CMAKE_PREFIX_PATH or set
  "gz-cmake3_DIR" to a directory containing one of the above files.  If
  "gz-cmake3" provides a separate development package or SDK, be sure it has
  been installed.


```

**Comment by traversaro** [comment]
> @traversaro  should I add the environment variable `GZ_SIM_RESOURCE_PATH` as it has been done for `GAZEBO_MODEL_PATH` to find the iCub and ErgoCub models for Gazebo Classic?

Yes. Eventually we will add gz-sim support there, so let's already do it so we can forget about this.

**Comment by traversaro** [comment]
Note that probably given all the problems related to co-installation of gz-sim and gazebo-classic we can skip the CI part for now.

File: cmake/RobotologySuperbuildOptions.cmake [hunk 1]
```diff
@@ -27,6 +27,10 @@ option(ROBOTOLOGY_USES_GAZEBO "Enable compilation of software that depends on Ga
 option(ROBOTOLOGY_USES_PCL_AND_VTK "Enable compilation of software that depends on PCL and VTK" OFF)
 option(ROBOTOLOGY_USES_MUJOCO "Enable compilation of mujoco and software that depends on it" OFF)
 
+## Enable packages that depend on the Gazebo simulator
```
Comment (traversaro) [inline cmake/RobotologySuperbuildOptions.cmake:30]:
```suggestion
## Enable packages that depend on the Modern Gazebo (gz-sim) simulator
```

File: doc/cmake-options.md [hunk 2]
```diff
@@ -17,7 +17,8 @@ Table of Contents
     * [Human Dynamics profile](#human-dynamics)
     * [Event-driven profile](#event-driven)
   * [Dependencies-specific documentation](#dependencies-specific-documentation)
-    * [Gazebo simulator](#gazebo)
+    * [Gazebo Classic simulator](#gazebo-classic-simulator)
+    * [Gazebo simulator](#gazebo-simulator)
```
Comment (traversaro) [inline doc/cmake-options.md:21]:
```suggestion
    * [Modern Gazebo (gz-sim) simulator](#gazebo-simulator)
```

File: doc/cmake-options.md [hunk 3]
```diff
@@ -193,13 +196,17 @@ This profile is enabled by the `ROBOTOLOGY_ENABLE_EVENT_DRIVEN` CMake option. Fo
...
 
-## Gazebo
+## Gazebo Classic simulator
 
 Support for this dependency is enabled by the `ROBOTOLOGY_USES_GAZEBO` CMake option, that enables the software that depends on "Classic Gazebo".
 
 ### Check the installation
 Follow the steps in  https://github.com/robotology/icub-models#use-the-models-with-gazebo to check if the Gazebo-based iCub simulation works fine.
 
+## Gazebo simulator
```
Comment (traversaro) [inline doc/cmake-options.md:206]:
```suggestion
## Modern Gazebo simulator
```

File: doc/cmake-options.md [hunk 4]
```diff
@@ -193,13 +196,17 @@ This profile is enabled by the `ROBOTOLOGY_ENABLE_EVENT_DRIVEN` CMake option. Fo
...
+
+Support for this dependency is enabled by the `ROBOTOLOGY_USES_GZ_SIM` CMake option, that enables the software that depends on "Gazebo".
```
Comment (traversaro) [inline doc/cmake-options.md:208]:
```suggestion
Support for this dependency is enabled by the `ROBOTOLOGY_USES_GZ_SIM` CMake option, that enables the software that depends on "Modern Gazebo" (gz-sim).
```

**Comment by xela-95** [comment]
> I left a bunch of comments, but the gist is the following: never use "Gazebo" without specification, it is going to be too confusing otherwise for users that called "Classic Gazebo" Gazebo for years.

Ok, what term we should use for new Gazebo? Modern Gazebo or gz-sim?

**Comment by xela-95** [comment]
I tested the usage of `ergocub-software` and `gz-sim-yarp-plugins` from the `robotology-superbuild` on Ubuntu 22.04 and WSL, both with [conda-forge](https://github.com/robotology/robotology-superbuild/blob/master/doc/conda-forge.md#source-installation)and [apt ](https://github.com/robotology/robotology-superbuild/tree/master?tab=readme-ov-file#linux-from-source-with-dependencies-provided-by-apt) dependencies. Both worked as expected.

**Comment by traversaro** [comment]
Thanks, I think we can proceed with merging. Existing conda users may need to pay attention as now `gazebo` needs to be installed manually and is not installed by the long list of dependencies in the README, but I guess it should be easy to manage as a change. @xela-95 can I squash?

**Comment by xela-95** [comment]
> Thanks, I think we can proceed with merging. Existing conda users may need to pay attention as now `gazebo` needs to be installed manually and is not installed by the long list of dependencies in the README, but I guess it should be easy to manage as a change. @xela-95 can I squash?

Yep exactly, now `gazebo` has to be installed manually also from conda, but I think this will not overcomplicate the workflow too much.

Sure, squeeze! 🧽

**Review by traversaro — APPROVED** [review]

**Pull request merged**